### ISSUE TITLE
No callback on disconnect when turning screen off

### ISSUE THREAD

**Issue by mcorner** [twilio/video-quickstart-objc#116]
### Description

If I press the power button (shutting the screen off) on the iOS device during a call, wait, and then turn it back on, the remote video freezes and never recovers.  But we also don't get a callback.

### Steps to Reproduce

1. Start a video chat
2. Press power button
3. Wait 5+ seconds.
4. turn the device back on.

#### Code

We have our own module, but I don't know which part of the code might be relevant.

#### Expected Behavior

didDisconnectWithError callback for the room would be one possibility.

Is there a better way to deal with this with lifecycle events?

#### Actual Behavior

Video freezes.  There are some error messages in Xcode's console, but no callbacks:

```
2018-05-24 14:01:05.012006-0400 OurApp[332:185861] ERROR:TwilioVideo:[Signaling]:RESIP::TRANSPORT: Socket was closed (due to backgrounding or handover), deleting connection ...
2018-05-24 14:01:05.012918-0400 OurApp[332:185906] ERROR:TwilioVideo:[Core]:I/O error connecting to sdkgw.us1.twilio.com:443, I/O error.
2018-05-24 14:01:09.447637-0400 OurApp[332:186346] dnssd_clientstub write_all(40) DEFUNCT
2018-05-24 14:01:09.484968-0400 OurApp[332:186346] dnssd_clientstub deliver_request ERROR: write_all(40, 61 bytes) failed
2018-05-24 14:01:09.485140-0400 OurApp[332:186346] dnssd_clientstub write_all(40) DEFUNCT
2018-05-24 14:01:20.357605-0400 OurApp[332:185861] ERROR:TwilioVideo:[Signaling]:RESIP::TRANSPORT: Socket was closed, reopening ...
```

#### Reproduces How Often

100%

#### Logs

See above

### Versions

See below
#### Video iOS SDK

2.0.1 via pods
#### Xcode

9.3.1

#### iOS Version

10.3.3

#### iOS Device

iPhone 5

**Comment by mcorner** [comment]
Oh.....I just found this setting in Info.plist for required background modes:
"App plays audio or streams audio/video using AirPlay"

This appears to fix my issue.  Can you confirm?  (And probably that should be documented...)

**Comment by paynerc** [comment]
Hey @mcorner,

You are correct, to have the signaling and media connections stay alive when the app is in the background, you need to set the `Audio, AirPlay, and Picture in Picture` background mode ticked for your app.

<img width="1976" alt="image" src="https://user-images.githubusercontent.com/25950/40551348-9d7d3a4c-600a-11e8-8a2f-2e3563ee714a.png">

I am checking the documentation to make sure that this is mentioned and if not, I will create a ticket to get that addressed.

Let me know if you have any further questions and thanks for reporting this!

Ryan

**Comment by paynerc** [comment]
I checked the documentation and you are correct, this is not noted anywhere. I am very sorry that that was missed and am surprised that it was never called out before.

I have created an internal ticket, ISDK-2014, which will address adding this information to the documentation.

Again, thank you for reporting this omission in the documentation. We should have that addressed very soon. One the docs have been updated and published, I will let you know and then close out this issue.

Ryan
